### PR TITLE
Contracts improvements and bugs fixing

### PR DESCRIPTION
**Pull request by szotov**
* Step-wise unlock only if the miner is active
* `receiveApproval` method in the `MinersEscrow `to deposit tokens with one transaction
* Draft of docs for contracts
* Minimum reward for policy
* Fixes for `finishUpgrade` method (`MinersEscrow` and `Issuer`)
* Vote method in the `UserEscrow`, all tokens will be unlocked after specified time

### CODE REVIEW

File: doc/contracts.rst [hunk 1]
```diff
@@ -0,0 +1,105 @@
+NuCypher contracts
+========================
+* `NuCypherToken` contract is the ERC20 token contract with additional function - burn own tokens (only for owners)
+* `MinersEscrow` contract holds Ursula's stake, stores information about Ursulas activity and assigns a reward for participating in NuCypher network. The `Issuer` contract is part of the `MinersEscrow` and uses only to split code
+* `PolicyManager` contract holds policies fee and distributes fee by periods
+* `Upgradeable` is base contract for upgrading (<nucypher.blockchain.eth/project/contracts/proxy/README.MD>)
+* `Dispatcher` contract is used as proxy to other contracts. This provides upgrading of the `MinersEscrow`, `PolicyManager` and `Government` contracts
+* `Government` contract holds ownership for dispatchers and provides voting for upgrading of those contracts
+* `UserEscrow` contract locks tokens for some time. In that period tokens are lineraly unlocked and all tokens can be used as a stake in the `MinersEscrow` contract
```
Comment (michwill) [inline doc/contracts.rst]: The `UserEscrow` should be changed from linear unlocking to unlocking all in the specified time. It should also simplify the contract

Comment (szotov) [reply doc/contracts.rst]: I changed the unlock. initialDeposit uses duration as input parameter, is it ok? Or should be end time?

Comment (michwill) [reply doc/contracts.rst]: What's the difference (what's the "end time"?)

Comment (szotov) [reply doc/contracts.rst]: I mean, endTime = blockTime + duration or endTime comes from parameter

**Pull request merged**